code/modules/admin/verbs/playsound.dm

```diff
@@ -151,10 +162,15 @@
 	if(web_sound_url && !findtext(web_sound_url, GLOB.is_http_protocol))
 		tgui_alert(user, "The media provider returned a content URL that isn't using the HTTP or HTTPS protocol. This is a security risk and the sound will not be played.", "Security Risk", list("OK"))
 		to_chat(user, span_boldwarning("BLOCKED: Content URL not using HTTP(S) Protocol!"), confidential = TRUE)
-
 		return
 	if(web_sound_url || stop_web_sounds)
-		for(var/m in GLOB.player_list)
+		var/list/players = GLOB.player_list
+		if (range)
+			players = list()
+			for (var/mob/listening_mob in get_hearers_in_view(range, user))
+				if (listening_mob.client)
+					players += listening_mob.client
```

Comment:
this doesnt work. the code runs on mobs, not clients.

Reply:
Yeah I just realised, I'm already pushing a fix